language-plutus-napkin/src/Language/PlutusCore/Type.hs

```diff
@@ -27,19 +30,32 @@ import           Language.PlutusCore.Identifier
 import           Language.PlutusCore.Lexer.Type
 import           PlutusPrelude
 
-data Name a = Name a BSL.ByteString Unique
-            deriving (Show, Generic, NFData)
+data Name a = Name { nameLoc :: a, asString :: BSL.ByteString, unique :: Unique }
+            deriving (Functor, Show, Generic, NFData)
+
+-- N.B. unfortunately this is necessary to allow the test suite to instead
```

Comment:
Can't we do it the other way around? That is, have the `Eq` instance use uniques for comparison and have a function `nameStringEquality` that compares strings and is used by the testsuite?

Reply:
I could use that in the test suite, but it would be a good amount of (straightforward) work.

As for Haddock comments - I've added them in places, but I still need a bit more as I am trying to avoid trivial examples.